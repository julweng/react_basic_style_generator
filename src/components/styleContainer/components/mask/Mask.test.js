import React from "react"
import { render, cleanup } from "@testing-library/react"
import "jest-dom/extend-expect"
import Mask from "./index"

const createProps = ({ loaded = false } = {}) => ({
  loaded
})

describe("Mask", () => {
  afterEach(cleanup)

  it("Should render without crashing", () => {
    const props = createProps()
    const { container } = render(<Mask {...props} />)
    expect(container).toBeTruthy()
    expect(container).toMatchSnapshot()
  })

  it("Should render without loaded class if loaded prop is false", () => {
    const props = createProps()
    const { container } = render(<Mask {...props} />)
    const mask = container.querySelector("span")
    expect(mask).not.toHaveClass("loaded")
  })

  it("Should render with loaded class if loaded prop is true", () => {
    const props = createProps({ loaded: true })
    const { container } = render(<Mask {...props} />)
    const mask = container.querySelector("span")
    expect(mask).toHaveClass("loaded")
  })
})
