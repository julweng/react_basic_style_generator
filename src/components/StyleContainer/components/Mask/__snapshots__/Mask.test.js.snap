// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mask Should render without crashing 1`] = `
<div>
  <span
    class="mask "
  />
</div>
`;
